### ISSUE TITLE
Interface for reading ruptures from sources to use with Hamlet

### ISSUE THREAD

**Issue by cossatot** [gem/oq-engine#7265]
Hamlet requires a list of ruptures from the source model logic tree, with optional filtering.

This could be done with a tool like csm2rup.py, but with some modifications.

The filtering needs to be done by SSM logic tree branch, by source type, tectonic region type, and by depth.  Obviously the first three can be done before the sources are expanded into ruptures.  Filtering by spatial region is done at a later step if required.

The returned attributes for each rupture are:
- lat
- lon
- depth
- mag
- strike
- dip
- rake
- occurrence_rate
- src_id (optional)
- rup_id (optional)

And this could be delivered as a dataframe.

Hamlet does not (currently) use any sites, and does not use a `job.ini` file, and I do not want to introduce those as a requirement so if there is a workaround for that (in order to build a `CompositeSourceModel` for instance) it would be nice.  It just reads the `ssmLT.xml` file (or whatever it may be named for that model).  I also don't know what involved in the rupture groups exactly but I don't think they correspond exactly to the attributes that I group or filter on, so I do not know if it is useful.

I would also like to keep things in memory where possible rather than relying on the datastore.

Returning a dict (or hdf5 file) with this format would be the best:

```python

ruptures = {
  'branch_1': {
    'ruptures': branch_1_rupture_dataframe, 
    'weight': branch_1_weight,
  },
  'branch_2': {
    'ruptures': branch_2_rupture_dataframe, 
    'weight': branch_2_weight,
  },
}
```
for all of the branches to be analyzed (and if a single branch, just a dictionary of that branch with weight 1.0).

Thanks, and I am happy to help but some guidance would be good to proceed most efficiently.

**Comment by cossatot** [comment]
@micheles thanks for this, it's helpful, and I have adapted some of it into Hamlet.  It does require a gmmLT.xml file, though--this breaks most of the tests and means that we can't use it during model development until that component has been made.  Is there a workaround here?

**Comment by micheles** [comment]
Yes, of course there is a workaround. Since at the end the GMPE will not be used by hamlet you can use any (for instance AbrahamsonEtAl2014 which is the first in lexicographic order). Just change the last two lines with
```python
   ...
    gsim='AbrahamsonEtAl2014',
    inputs=dict(source_model_logic_tree='ssmLT.xml'))
```
